File: domain/src/main/java/com/beat/domain/user/repository/UserRepository.java
package com.beat.domain.user.repository;

import java.util.List;
import java.util.Optional;

import com.beat.domain.user.domain.Users;

public interface UserRepository {
	Optional<Users> findById(Long id);

	List<Users> findAll();

	Users save(Users users);

	void delete(Users users);
}

File: core/domain/src/main/kotlin/com/beat/domain/user/repository/UserRepository.kt
package com.beat.domain.user.repository

import com.beat.domain.user.model.Users
import java.util.*

@JvmSuppressWildcards
interface UserRepository {
    fun findById(id: Long?): Optional<Users>

    fun findAll(): List<Users>

    fun save(users: Users): Users

    fun delete(users: Users)
}
